src/Listing/Bid/Bid.router.js

```diff
@@ -82,17 +101,69 @@ export class BidRouter {
       bids = await Bid.findByAssetId(id, assetType)
     }
 
-    return sanitizeBids(bids)
+    try {
+      const bidder = server.extractFromReq(req, 'bidder')
+      bids = bids.filter(bid => bid.bidder === bidder)
+    } catch (errror) {
+      // Do nothing
+    }
+
+    try {
+      const sanitize = server.extractFromReq(req, 'sanitize')
+      return sanitize === 'false' ? bids : sanitizeBids(bids)
+    } catch (errror) {
+      return sanitizeBids(bids)
+    }
   }
 
-  async getBid(req) {
+  async getBidById(req) {
     const id = server.extractFromReq(req, 'id')
     const bid = await Bid.findOne({ id })
 
     if (!bid) {
       throw new Error(`Could not find a valid bid with the id: "${id}"`)
     }
 
-    return sanitizeBids(bid)
+    return bid
+  }
+
+  async getAddressBids(req) {
+    const address = server.extractFromReq(req, 'address')
+    let bids = []
+
+    try {
+      const status = server.extractFromReq(req, 'status')
+      bids = await Bid.findByAddressAndStatus(address, status)
+    } catch (error) {
+      bids = await Bid.findByAddress(address)
+    }
+
+    return sanitizeBids(bids)
+  }
+
+  async getAssetBidded(req) {
+    const reqQueryParams = new ReqQueryParams(req)
+
+    const opt = {}
+    if (reqQueryParams.has('bidder')) {
+      opt['bidder'] = reqQueryParams.get('bidder')
+    }
+
+    if (reqQueryParams.has('status')) {
+      opt['status'] = reqQueryParams.get('status')
+    }
+
+    let assets
+
+    try {
+      const status = server.extractFromReq(req, 'status')
+
+      assets = await Bid.findBiddedAssetsByStatus(opt, status)
+    } catch (error) {
+      console.log(error.message)
+      // assets = await Bid.findBiddedAssets(opt)
+    }
+
+    return sanitizeAssets(assets)
```

Comment:
**General file question**:

I don't get why are we having sanitize as an argument for these endpoints. Granted I'm reading this top to bottom so maybe reading the front-end part will illuminate me, but can you explain?
Why not returning everything always? We never needed this before